### DISCUSSION TITLE
Event Enrollment Object.c  is not supported ?

### DISCUSSION THREAD

**Discussion by maimomohali** [bacnet-stack/bacnet-stack#599] [Q&A] [answered]
In implementing Backnet Server Task,  I found  that   file about Event enrollment objec is not exist in path /src/bacnet/basic/object/
I thought event Enrollment Object would exist in the /src/bacnet/basic/object/path just like any obj -  trendlog obj,  callendar obj, notification class obj...
Do I have to write code of event_enrollment.c  myself    ??,    I think this problem is  from my misunderstanding about  concept of  evnet enrollment  object, 
What is event_Enrollment Obj and how should it be implemented?

**Comment by skarg** [comment, marked as answer]
At the moment there isn't a basic example of the Event Enrollment object in this BACnet Stack library.

See the BACnet standard - SSPC 135 - for the specification and description of the Event Enrollment object. There is a free preview online: 
https://www.ashrae.org/technical-resources/standards-and-guidelines
Scroll down to 135-2020 and follow the link.
```
12.12 Event Enrollment Object Type
The Event Enrollment object type defines a standardized object that represents and contains the information required for
algorithmic reporting of events. For the general event concepts and algorithmic event reporting, see Clause 13.2.

For the Event Enrollment object, detecting events is accomplished by performing particular event and fault algorithms on
monitored values of a referenced object. The parameters for the algorithms are provided by the Event Enrollment object. The
standard event algorithms are defined in Clause 13.3. The standard fault algorithms are defined in Clause 13.4. Event
Enrollment objects do not modify or otherwise influence the state or operation of the referenced object.

For the reliability indication by the Reliability property of the Event Enrollment object, internal unreliable operation such as
configuration error or communication failure takes precedence over reliability indication for the monitored object. Fault
indications determined by the fault algorithm, if any, have least precedence.

Clause 13.2 describes the interaction between Event Enrollment objects, the Notification Class objects, and the Alarm and
Event application services.
```
